### PR TITLE
feat: add unavailablebalance api

### PR DESCRIPTION
**Pull request by isaac-fibonacci**
## What does this MR do?

Add UnavailableBalance API

## Check list  

- [X] All Merge Request pipelines should be passed.
- [X] Make sure all the changes are necessary and correct.
- [X] Make sure no needed change missing (all necessary changes are made).
- [X] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [] All necessary unit tests are written.
- [X] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: src/presentation/http/controllers/balance.controller.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const data: IOrderRepo.getUnbroadcastedInput = {
+      network,
+      userAddress: user_address,
+      assetSymbol: req.query.assetSymbol as string,
+    };
+    const result = await this.dataService.getUserUnbroadcasted(data);
+    res.send({
+      success: true,
+      data: {
+        assetSymbol: data.assetSymbol,
```
Comment (hypersolo) [inline src/presentation/http/controllers/balance.controller.ts:25]: nice job uncle :heart: 
we use snake_case in all responses, please use it here too :clinking_glasses:

File: src/domain/ports/data-service.interface.ts [hunk 2]
```diff
@@ -66,6 +68,9 @@ export namespace IDataService {
 
   export interface schema {
     getUserOrders(params: getUserOrdersInput): Promise<Order[]>;
+    getUserUnbroadcasted(
+      params: IOrderRepo.getUnbroadcastedInput,
```
Comment (hypersolo) [inline src/domain/ports/data-service.interface.ts:72]: please don't use IOrderRepo interface here :tulip:

**Comment by hypersolo** [comment]
use .toUppercase for asset_symbol 🤞

File: src/presentation/http/validators/http.validator.ts [hunk 3]
```diff
@@ -87,6 +87,7 @@ export class HTTPValidator {
     ],
     type: (location: Location) => [location("type").isIn(["limit", "market"])],
     id: (location: Location) => [location("id").isInt({ min: 0 })],
+    assetSymbol: (location: Location) => [location("assetSymbol").isString()],
```
Comment (WilliamHawkingStephen) [inline src/presentation/http/validators/http.validator.ts:90]: nice job uncle ❤️
- also these http validators all of them are snake case

File: src/presentation/http/controllers/balance.controller.ts (hunk 1 above)
Comment (isaac-fibonacci) [reply src/presentation/http/controllers/balance.controller.ts:25]: done

File: src/presentation/http/validators/http.validator.ts (hunk 3 above)
Comment (isaac-fibonacci) [reply src/presentation/http/validators/http.validator.ts:90]: done

File: src/domain/ports/data-service.interface.ts (hunk 2 above)
Comment (isaac-fibonacci) [reply src/domain/ports/data-service.interface.ts:72]: done

**Review by hypersolo — APPROVED** [review]

**Review by WilliamHawkingStephen — APPROVED** [review]

**Pull request merged**